src/runtime/native/runtime.ml

```diff
@@ -101,24 +124,43 @@ let dump_counters_exn =
   Common.write_runtime_data
 
 let dump () =
-  match file_channel () with
-  (* Failure "TypeError: runtime.unix_open is not a function" indicates that the
-     code was compiled by js_of_ocaml. In this case, we *want* to do nothing at
-     process exit. See https://github.com/aantron/bisect_ppx/issues/334. *)
-  | exception Failure _ -> ()
-  | None -> ()
-  | Some channel ->
-      (try
-        dump_counters_exn channel
-      with _ ->
-        verbose Unable_to_write_file);
-      close_out_noerr channel
+  if not !bisect_file_written then
+    (if !sigterm_enable then
+       ignore @@ Unix.(sigprocmask SIG_BLOCK [Sys.sigterm]) ;
```

Comment:
Blocking of the signal should be moved to a wrapper that will be called only during `at_exit`. As this code is now, the signal will be blocked even if `dump` is called from the SIGTERM handler. This is not necessary, as SIGTERM should already be blocked during the SIGTERM handler.

I also suggest to unblock SIGTERM at the end of the new wrapper around `dump`, so that if there are other `at_exit` handlers in the binary under test, they can be interrupted by SIGTERM as normal.

Reply:
Also, I think you can stick with `Sys.signal` here, rather than using `Unix.sigprocmask`.

Reply:
In 0f87a8f I unblock SIGTERM after `at_exit`. I also add a (convoluted) test that verifies that succeeding at_exit hooks can be interrupted.